lib/modules/manager/cargo/artifacts.ts

```diff
@@ -51,9 +51,23 @@ async function cargoUpdatePrecise(
       ` --manifest-path ${quote(manifestPath)} --workspace`,
   ];
 
+  const anyNonLockfileUpdate = updatedDeps.some(
+    (dep) => dep.updateType !== 'lockfileUpdate',
+  );
+  if (anyNonLockfileUpdate) {
+    cmds.push(
+      `cargo fetch --config net.git-fetch-with-cli=true` +
```

Comment:
Should we do cargo fetch _instead_ of cargo update?

Reply:
I changed the implementation to run `cargo update --workspace` since it appears to be updating the lockfile in the same way without actually needing to fetch the dependencies.